test/tools/libtesteth/TestSuite.cpp

```diff
@@ -55,24 +55,36 @@ void removeComments(json_spirit::mValue& _obj)
 	}
 }
 
-void addClientInfo(json_spirit::mValue& _v, fs::path const& _testSource, h256 const& _testSourceHash)
+void addClientInfo(json_spirit::mValue& _testDestination, json_spirit::mValue const& _testSource, fs::path const& _testSourcePath, h256 const& _testSourceHash)
 {
-	for (auto& i: _v.get_obj())
+	for (auto& i: _testDestination.get_obj())
 	{
 		json_spirit::mObject& o = i.second.get_obj();
 		json_spirit::mObject clientinfo;
 
 		string comment;
-		if (o.count("_info"))
+		if (o.count("_info")) // take the comment from generated test file
 		{
-			json_spirit::mObject& existingInfo = o["_info"].get_obj();
-			if (existingInfo.count("comment"))
-				comment = existingInfo["comment"].get_str();
+			json_spirit::mObject const& info = o["_info"].get_obj();
+			comment = info.count("comment") ? info.at("comment").get_str() : string();
+		}
+
+		// copy comment from test source if there is one
+		for (auto& j: _testSource.get_obj())
+		{
+			json_spirit::mObject const& o2 = j.second.get_obj();
+			if (o2.count("_info") && j.first == i.first)
+			{
+				json_spirit::mObject const& existingInfo = o2.at("_info").get_obj();
+				if (existingInfo.count("comment"))
+					comment = existingInfo.at("comment").get_str();
+				break;
```

Comment:
Why is the first comment taken, and the rest is ignored?

Reply:
the comment from exact tetst is taken.   
the loop is for if  (info exist and test name is the same) 
searching for the test in filler file

Reply:
I think we don't need to find the test again for this purpose.  When we are filling the test, we have access to the original filler and the generated test.  Let's copy comments there.

Reply:
OK, maybe the current form is the easiest for now.

Reply:
we dont really have an access 
json_spirit::mValue output = doTests(v, true);
addClientInfo(output, v, boostRelativeTestPath, sha3(byteContents));

first all tests are being executed then we add comments to all tests

Reply:
to do this as you say we better copy the info field when filling the test 
then restore the original logic as it was